wordpress-mobile/WordPress-iOS
# Adds testing to the QRLogin code

### Description

Adds unit testing and changes related to making the code more testable:

- Extracts quite a bit of logic into protocols to make some areas of the code more testable with dependency injection
- A notable example is removing the camera session logic from the scanning coordinator and moving it into its own class adhering to the `QRCodeScanningSession` protocol
   - The reason for this is because the AVDevice logic isn't able to be tested without using a real device, and this allows me to create a mock object to test with

- All protocols that were defined are in their own file `QRLoginProtocols` to clean things up a bit
- Moved some files into folders to clean the project list up

### To test:
- Verify tests are 🟢 
- Test using the steps from the [previous PRs](https://github.com/wordpress-mobile/WordPress-iOS/pulls?q=is%3Apr+author%3Aemilylaguna+is%3Aclosed+label%3A%22QR+Login%22):  

## Regression Notes
1. Potential unintended areas of impact
TBD

2. What I did to test those areas of impact (or what existing automated tests I relied on)
TBD

3. What automated tests I added (or what prevented me from doing so)
TBD

PR submission checklist:
